astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -41,13 +43,21 @@ class galactocentric_frame_defaults(ScienceState):
     ``Galactocentric`` or ``Galactocentric()`` with no explicit arguments.
 
     This class controls the parameter settings by specifying a string name,
-    which can be one of:
+    which has the pre-specified options:
 
     - 'pre-v4.0': The current default value, which sets the default frame
       attribute values to their original (pre-astropy-v4.0) values.
     - 'v4.0': The attribute values as updated in Astropy version 4.0.
     - 'latest': An alias of the most recent parameter set (currently: 'v4.0')
 
+    Alternatively, user-defined parameter settings may be registered and set.
+
+    This class also tracks the references for all parameter values in the
+    attribute ``_references``, as well as any further information in
```

Comment:
Seeing this makes me realize this was always a bit weird: we should *never* expose private symbols as public API! I suggest we change this to plain attributes. Or, to keep backward compatibility, store the references and state in the private attributes, but expose them as read-only `@classproperty`.

Reply:
I agree. The `_state` attribute could coordinate all the information

```python
class ScienceState...
    ...
    @classproperty
    def value(cls):
         return cls._state["value"]
    
    @classproperty
    def _value(cls):  # for backward compatibility
         return cls._state["value"]

    @classproperty
    def references(cls):
         return cls._state["references"]
    
    @classproperty
    def _references(cls):  # for backward compatibility
         return cls._state["references"]
```

Reply:
Scratch the above. Unfortunately, class properties cannot be used on attributes when there is no instance of the class. This is why a metaclass is necessary as only a metaclass (or a modification to how `__dict__` functions) can create properties that work on non-instantiated classes.

as example
```python
class A:
    _value = 2

    @property
    @classmethod
    def value(cls):
        return cls.value
```

`A.value` is as a property object, not `2` until an instance of A is created.

```python
class AMeta(type):
    @property
    @classmethod
    def value(cls):
        return cls.value

class A(metaclass=AMeta):
    _value = 2
```

Reply:
Are you sure? We do have https://docs.astropy.org/en/latest/api/astropy.utils.decorators.classproperty.html#astropy.utils.decorators.classproperty for this purpose (which is what I was referring to)

Reply:
I stand corrected. That would work. They are read-only, which is probably an advantage given that ``set`` is desired means to change the state.